client/my-sites/checkout/checkout/composite-checkout.js

```diff
@@ -654,56 +665,138 @@ function getCheckoutEventHandler( dispatch ) {
 				return dispatch(
 					recordTracksEvent( 'calypso_checkout_composite_step_changed', { step: action.payload } )
 				);
-			case 'STRIPE_TRANSACTION_BEGIN':
+			case 'STRIPE_TRANSACTION_BEGIN': {
+				dispatch(
+					recordTracksEvent( 'calypso_checkout_form_submit', {
+						credits: null,
+						payment_method: 'WPCOM_Billing_Stripe_Payment_Method',
+					} )
+				);
 				return dispatch(
 					recordTracksEvent( 'calypso_checkout_composite_stripe_submit_clicked', {} )
 				);
-			case 'STRIPE_TRANSACTION_ERROR':
+			}
+			case 'STRIPE_TRANSACTION_ERROR': {
+				dispatch(
+					recordTracksEvent( 'calypso_checkout_payment_error', {
+						error_code: null,
+						reason: String( action.payload ),
+					} )
+				);
 				return dispatch(
 					recordTracksEvent( 'calypso_checkout_composite_stripe_transaction_error', {
 						error_message: String( action.payload ),
 					} )
 				);
-			case 'PAYPAL_TRANSACTION_BEGIN':
+			}
+			case 'PAYPAL_TRANSACTION_BEGIN': {
+				dispatch( recordTracksEvent( 'calypso_checkout_form_redirect', {} ) );
+				dispatch(
+					recordTracksEvent( 'calypso_checkout_form_submit', {
+						credits: null,
```

Comment:
🤔 We might be able to get this out of the line items and pass it along with this event... but maybe it's not crucial.